sys/riotboot/flashwrite.c

```diff
@@ -104,6 +158,10 @@ int riotboot_flashwrite_finish_raw(riotboot_flashwrite_t *state,
 
     uint8_t *slot_start = (uint8_t *)riotboot_slot_get_hdr(state->target_slot);
 
+#if CONFIG_RIOTBOOT_FLASHWRITE_RAW
+    memcpy(state->firstblock_buf, bytes, len);
+    flashpage_write_raw(slot_start, state->firstblock_buf, RIOTBOOT_FLASHPAGE_BUFFER_SIZE);
+#else
```

Comment:
```suggestion
    if (CONFIG_RIOTBOOT_FLASHWRITE_RAW) {
        memcpy(state->firstblock_buf, bytes, len);
        flashpage_write_raw(slot_start, state->firstblock_buf, RIOTBOOT_FLASHPAGE_BUFFER_SIZE);
    } else {
```

Reply:
`state->firstblock_buf` is not available when `CONFIG_RIOTBOOT_FLASHWRITE_RAW`, guarding with an `if` will result in compile errors when flashwrite raw is not used.